arcgis-ios-sdk-samples/Layers/Select ENC features/SelectENCFeaturesViewController.swift

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class SelectENCFeaturesViewController: UIViewController {
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemapStyle: .arcGISOceans)
+            mapView.touchDelegate = self
+            mapView.callout.isAccessoryButtonHidden = true
+        }
+    }
+    
+    /// The ENC layer that contains the current selected feature.
+    var currentENCLayer: AGSENCLayer?
+    /// A reference to the cancelable identify layer operation.
+    var identifyOperation: AGSCancelable?
+    
+    /// A URL to the temporary SENC data directory.
+    let temporaryURL: URL = {
+        let directoryURL = FileManager.default.temporaryDirectory.appendingPathComponent(ProcessInfo().globallyUniqueString)
+        // Create and return the full, unique URL to the temporary folder.
+        try? FileManager.default.createDirectory(at: directoryURL, withIntermediateDirectories: true)
+        return directoryURL
+    }()
+    
+    /// Load the ENC dataset and add it to the map view.
+    /// - Parameter mapView: The map view managed by the view controller.
+    func addENCExchangeSet(mapView: AGSMapView) {
+        let map = mapView.map!
+        // Load catalog file in ENC exchange set from bundle.
+        let catalogURL = Bundle.main.url(
+            forResource: "CATALOG",
+            withExtension: "031",
+            subdirectory: "ExchangeSetWithoutUpdates/ENC_ROOT"
+        )!
```

Comment:
It's case by case

- For App Store release, this won't happen because during archiving the app, all bundle resources (local, on-demand) will be downloaded included in the app archive. Then these files will be uploaded to Apple's server (which we can trust their uptime.)
- For developers building locally, it might be a problem but could be rare - unless ArcGIS server falsely reports the file is downloaded, but gives a corrupted/empty file instead. So I neither think this would be a problem
- If the portal item is removed, I haven't tried but assume it would be caught by the download script during build time
- It's been like this in older samples, so I think it should fine. e.g.
https://github.com/Esri/arcgis-runtime-samples-ios/blob/68f97e7bd3eee4acb1a4ff638ee34718a50b883a/arcgis-ios-sdk-samples/Scenes/Open%20mobile%20scene%20(scene%20package)/OpenMobileSceneViewController.swift#L25

Reply:
Thanks for your detailed response! Feel free to request more reviewers